### PR TITLE
improve the filtering of Actions in Staking section especially for de…

### PR DESCRIPTION
**Pull request by marc-aurele-besner**
## improve the filtering of Actions in Staking section especially for deregistered op

___

### **PR Type**
Enhancement

___

### **Description**
- Enhanced the filtering of actions in the Staking section to exclude the `Nominating` action for deregistered operators.
- Modified the `excludeActions` array dynamically based on the operator's status in `NominationManagement`, `OperatorManagement`, and `OperatorsList` components.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>NominationManagement.tsx</strong><dd><code>Enhance action filtering for deregistered operators in </code><br><code>NominationManagement</code></dd></summary>
<hr>

explorer/src/components/Operator/NominationManagement.tsx

<li>Added logic to exclude <code>Nominating</code> action for deregistered operators.<br> <li> Modified the <code>excludeActions</code> array dynamically based on operator <br>status.<br>

</details>

  </td>
  <td><a href="https://github.com/subspace/astral/pull/726/files#diff-efeab9f927b13d2314b3f2712da012a7368455850b73a81f40c4ea18408a49d0">+26/-17</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>OperatorManagement.tsx</strong><dd><code>Enhance action filtering for deregistered operators in </code><br><code>OperatorManagement</code></dd></summary>
<hr>

explorer/src/components/Operator/OperatorManagement.tsx

<li>Added logic to exclude <code>Nominating</code> action for deregistered operators.<br> <li> Modified the <code>excludeActions</code> array dynamically based on operator <br>status.<br>

</details>

  </td>
  <td><a href="https://github.com/subspace/astral/pull/726/files#diff-1728839c747fa9b636fe04951d7342836b3786430d6640d8bd9bd9ca72e9c8e3">+19/-12</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>OperatorsList.tsx</strong><dd><code>Enhance action filtering for deregistered operators in OperatorsList</code></dd></summary>
<hr>

explorer/src/components/Operator/OperatorsList.tsx

<li>Added logic to exclude <code>Nominating</code> action for deregistered operators.<br> <li> Removed redundant condition check for nominator.<br>

</details>

  </td>
  <td><a href="https://github.com/subspace/astral/pull/726/files#diff-2144a16cae24f1edb0774bfb0c1d2db81bf6100af1920189a1b1dbda4506bdaf">+1/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

**Comment by jfrank-summit** [comment]
When slashed there shouldn't be any options.
image

**Comment by marc-aurele-besner** [comment]
> When slashed there shouldn't be any options. image

I did the change for the operator list, but when slashed, do the nominators or operators can still withdraw, or everything taken?

**Comment by jfrank-summit** [comment]
> > When slashed there shouldn't be any options. image
> 
> I did the change for the operator list, but when slashed, do the nominators or operators can still withdraw, or everything taken?

Everything is taken.

File: explorer/src/components/Operator/OperatorsList.tsx
Comment (jfrank-summit) [inline explorer/src/components/Operator/OperatorsList.tsx:1]: The action dropdown is still showing up on the preview for me.

Comment (marc-aurele-besner) [reply explorer/src/components/Operator/OperatorsList.tsx:1]: It should be fix now
image

**Review by jfrank-summit — CHANGES_REQUESTED** [review]

File: explorer/src/components/Operator/NominationManagement.tsx [hunk 1]
```diff
@@ -246,6 +246,12 @@ export const NominationManagement: FC = () => {
               .deregistered
           )
             excludeActions.push(OperatorActionType.Nominating)
+          if (
+            row.original.operator.status &&
+            (JSON.parse(row.original.operator.status) as unknown as { slashed: object }).slashed ===
+              null
+          )
```
[the remark this answers is not in the document]
Comment (marc-aurele-besner) [reply explorer/src/components/Operator/NominationManagement.tsx]: yes it work, I just made a ticket to merge all the staking page together (except register) so after this PR I'll merge them all

**Review by jfrank-summit — APPROVED** [review]

**Pull request merged**